feat: Add info about plugin remapping and update paperweight version. (#366)
PaperMC/docs: docs/paper/dev/getting-started/userdev.mdx

types, names, and fields during development, and then remaps your plugin, so it can still be used with the obfuscated
server.

:::caution[1.20.5]

> As of Minecraft version 1.20.5, Paper ships with a Mojang-mapped runtime instead of reobfuscating the server to Spigot mappings as we have traditionally done since Paper 1.17.

Reviewer: I'm not sure the since 1.17 part is relevant. userdev users don't need to know or care about Paper's internal development process. Might add more confusion without any real benefit.

> As of Minecraft version 1.20.5 Paper will be shipped with a Mojang-mapped runtime instead of reobfuscating the server to Spigot mappings as we have traditionally done since Paper 1.17.

Reviewer 2: ```suggestion
As of Minecraft version 1.20.5, Paper will be shipped with a Mojang-mapped runtime instead of reobfuscating the server to Spigot mappings as we have traditionally done since Paper 1.17.
```

> See [Mojang-Mappings](#1205) for more details.

Reviewer 2: ```suggestion
See [here](#1205) for more details.
```

}
```

## 1.20.5

> As of 1.20.5 Paper will be shipping with a Mojang-mapped runtime instead of reobfuscating the server to Spigot mappings as we have traditionally done since Paper 1.17.

Reviewer 2: ```suggestion
As of 1.20.5, Paper will be shipping with a Mojang-mapped runtime instead of reobfuscating the server to Spigot mappings as we have traditionally done since Paper 1.17.
```

also, hasn't the server been Spigot-mapped ever since Paper was created, not only since 1.17?

> Additionally. CraftBukkit classes will no longer be relocated into a versioned package.

Reviewer 2: ```suggestion
Additionally, CraftBukkit classes will no longer be relocated into a versioned package.
```

This requires plugins to be deobfuscated before loading when necessary.

Most of this process is done automatically by paperweight but there are some important things to know when using NMS from now on.

> ### Default Mappings Assumption

Reviewer 2: ```suggestion
### Default mappings assumption
```

* By default, all Spigot/Bukkit plugins will be assumed to be Spigot-mapped if they do not specify their mappings namespace in the manifest. Conversely, all Paper plugins will be assumed to be Mojang-mapped if they do not specify their mappings namespace in the manifest.
* Spigot-mapped plugins will need to be deobfuscated on first load, Mojang-mapped plugins will not.

> If you want to map your pluign to Mojang-mappings you need to add the following code to your build script:

Reviewer 2: ```suggestion
If you want to map your plugin with Mojang mappings, you need to add the following code to your build script:
```

> If you want to map your plugin to Mojang-mappings you need to add the following code to your build script:

Reviewer 2: ```suggestion
If you want to map your plugin with Mojang mappings, you need to add the following code to your build script:
```

:::note

If you still want to target Spigot, you do not have to change anything.

:::

> ```kotlin title="build.gadle.kts"

Reviewer 2: ```suggestion
```kotlin title="build.gradle.kts"
```

paperweight.reobfArtifactConfiguration = io.papermc.paperweight.userdev.ReobfArtifactConfiguration.MOJANG_PRODUCTION
```

> additionally you need to remove all `dependsOn(reobfJar)` lines.

Reviewer 2: ```suggestion
Additionally, you need to remove all `dependsOn(reobfJar)` lines.
```

}
```

> ## 1.20.5

Reviewer 3: Newlines after headings please. Same with subsequent Headings
